dace/transformation/passes/dead_dataflow_elimination.py

```diff
@@ -159,8 +159,10 @@ def apply_pass(self, sdfg: SDFG, pipeline_results: Dict[str, Any]) -> Optional[D
                                             for code in leaf.src.code.code:
                                                 ast_find.generic_visit(code)
                                         except astutils.NameFound:
-                                            # then add the hint expression 
-                                            leaf.src.code.code = ast.parse(f'{leaf.src_conn}: dace.{ctype.to_string()}\n').body + leaf.src.code.code
+                                            # then add the hint expression
+                                            type_string = "bool_" if ctype.dtype == dtypes.bool else ctype.to_string()
+                                            leaf.src.code.code = ast.parse(
+                                                f'{leaf.src_conn}: dace.{type_string}\n').body + leaf.src.code.code
```

Comment:
This is not the way to go IMO. There should be a more general solution to generate Python types in the tasklet code.